x/superfluid/client/cli/tx.go

```diff
@@ -402,3 +405,92 @@ func NewCmdUnPoolWhitelistedPool() *cobra.Command {
 	flags.AddTxFlagsToCmd(cmd)
 	return cmd
 }
+
+// NewCmdUpdateUnpoolWhitelistProposal defines the command to create a new update unpool whitelist proposal command.
+func NewCmdUpdateUnpoolWhitelistProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-unpool-whitelist [flags]",
+		Args:  cobra.ExactArgs(0),
+		Short: "Update unpool whitelist proposal",
+		Long: "This proposal will update the unpool whitelist if passed. " +
+			"Every pool id must be valid. If proposal passes, and the pool id is invalid, the proposal will not take any effect. " +
```

Comment:
Why don't we run GetPoolAndPoke in ValidateBasic to prevent this from happening?

Reply:
Unfortunately, we don't have access to the gamm keeper in `ValidateBasic()`.

However, I found out that gov module attempts to test-run the proposal with cache context before allowing it to be submitted:
https://github.com/osmosis-labs/cosmos-sdk/blob/870132d311d162fdbe0e92fcc0a42ee3889afd0d/x/gov/keeper/proposal.go#L22-L29

So if pool doesn't exist, it will not be submitted. I will change the wording